File: FixedstaticArray.java
import java.util.Scanner;

public class FixedstaticArray {
    static String[] nim = new String[10];
    static String[] nama = new String[10];
    static int count = 0;
    static Scanner input = new Scanner(System.in);
    
    public static void main(String[] args) {
        System.out.println("=== PROGRAM ARRAY MAHASISWA ===");
        
        int pilih;
        do {
            System.out.println("\n=== MENU ===");
            System.out.println("1. Insert at beginning");
            System.out.println("2. Insert at given position");
            System.out.println("3. Insert at end");
            System.out.println("4. Delete from beginning");
            System.out.println("5. Delete given position");
            System.out.println("6. Delete from end");
            System.out.println("7. Delete first occurence");
            System.out.println("8. Show data");
            System.out.println("9. Exit");
            System.out.print("Pilih: ");
            
            pilih = input.nextInt();
            input.nextLine();
            
            switch(pilih) {
                case 1: insertAwal(); break;
                case 2: insertPosisi(); break;
                case 3: insertAkhir(); break;
                case 4: deleteAwal(); break;
                case 5: deletePosisi(); break;
                case 6: deleteAkhir(); break;
                case 7: deletePertama(); break;
                case 8: showData(); break;
                case 9: System.out.println("Program selesai"); break;
                default: System.out.println("Pilihan salah!");
            }
        } while(pilih != 9);
    }
    
    // 1. Insert at beginning
    static void insertAwal() {
        if(count >= 10) {
            System.out.println("Array penuh!");
            return;
        }
        
        System.out.print("NIM: ");
        String n = input.nextLine();
        System.out.print("Nama: ");
        String nm = input.nextLine();
        
        if(nim[0] == null) {
            count++;
        }
        
        nim[0] = n;
        nama[0] = nm;
        System.out.println("Data ditambahkan di No. 1");
        System.out.println("Count: " + count);
    }
    
    // 2. Insert at given position
    static void insertPosisi() {
        if(count >= 10) {
            System.out.println("Array penuh!");
            return;
        }
        
        System.out.print("No. (1-10): ");
        int no = input.nextInt();
        input.nextLine();
        
        if(no < 1 || no > 10) {
            System.out.println("No. harus 1-10!");
            return;
        }
        
        System.out.print("NIM: ");
        String n = input.nextLine();
        System.out.print("Nama: ");
        String nm = input.nextLine();
        
        int idx = no - 1;
        
        if(nim[idx] == null) {
            count++;
        }
        
        nim[idx] = n;
        nama[idx] = nm;
        System.out.println("Data ditambahkan di No. " + no);
        System.out.println("Count: " + count);
    }
    
    // 3. Insert at end
    static void insertAkhir() {
        if(count >= 10) {
            System.out.println("Array penuh!");
            return;
        }
        
        System.out.print("NIM: ");
        String n = input.nextLine();
        System.out.print("Nama: ");
        String nm = input.nextLine();
        
        if(nim[9] == null) {
            count++;
        }
        
        nim[9] = n;
        nama[9] = nm;
        System.out.println("Data ditambahkan di No. 10");
        System.out.println("Count: " + count);
    }
    
    // 4. Delete from beginning
    static void deleteAwal() {
        if(nim[0] == null) {
            System.out.println("No. 1 kosong!");
            return;
        }
        
        System.out.println("Menghapus No. 1: " + nim[0] + " - " + nama[0]);
        nim[0] = null;
        nama[0] = null;
        count--;
        System.out.println("Data dihapus");
        System.out.println("Count: " + count);
    }
    
    // 5. Delete given position
    static void deletePosisi() {
        if(count == 0) {
            System.out.println("Array kosong!");
            return;
        }
        
        System.out.print("No. yang dihapus (1-10): ");
        int no = input.nextInt();
        input.nextLine();
        
        if(no < 1 || no > 10) {
            System.out.println("No. harus 1-10!");
            return;
        }
        
        int idx = no - 1;
        
        if(nim[idx] == null) {
            System.out.println("No. " + no + " kosong!");
            return;
        }
        
        System.out.println("Menghapus No. " + no + ": " + nim[idx] + " - " + nama[idx]);
        nim[idx] = null;
        nama[idx] = null;
        count--;
        System.out.println("Data dihapus");
        System.out.println("Count: " + count);
    }
    
    // 6. Delete from end
    static void deleteAkhir() {
        if(nim[9] == null) {
            System.out.println("No. 10 kosong!");
            return;
        }
        
        System.out.println("Menghapus No. 10: " + nim[9] + " - " + nama[9]);
        nim[9] = null;
        nama[9] = null;
        count--;
        System.out.println("Data dihapus");
        System.out.println("Count: " + count);
    }
    
    // 7. Delete first occurence
    static void deletePertama() {
        if(count == 0) {
            System.out.println("Array kosong!");
            return;
        }
        
        System.out.print("Masukkan NIM yang akan dihapus: ");
        String cari = input.nextLine();
        
        for(int i = 0; i < 10; i++) {
            if(nim[i] != null && nim[i].equals(cari)) {
                System.out.println("Data ditemukan di No. " + (i+1) + ": " + nim[i] + " - " + nama[i]);
                nim[i] = null;
                nama[i] = null;
                count--;
                System.out.println("Data dihapus");
                System.out.println("Count: " + count);
                return;
            }
        }
        
        System.out.println("Data tidak ditemukan!");
    }
    
    // 8. Show data - dengan format "No." dan titik
    static void showData() {
        System.out.println("\n=== DATA MAHASISWA ===");
        System.out.println("Jumlah: " + count + " data");
        
        if(count == 0) {
            System.out.println("Tidak ada data");
            return;
        }
        
        System.out.println("\nNo.\tNIM\tNama");
        System.out.println("------------------------");
        
        for(int i = 0; i < 10; i++) {
            if(nim[i] != null) {
                System.out.println((i+1) + ".\t" + nim[i] + "\t" + nama[i]);
            }
        }
    }
}

File: FixedstaticArray.py
class FixedstaticArray:
    def __init__(self):
        self.nim = [None] * 10
        self.nama = [None] * 10
        self.count = 0
    
    def insert_awal(self):
        """1. Insert at beginning - selalu timpa index 0"""
        if self.count >= 10:
            print("Array penuh!")
            return
        
        n = input("NIM: ")
        nm = input("Nama: ")
        
        if self.nim[0] is None:
            self.count += 1
        
        self.nim[0] = n
        self.nama[0] = nm
        print("Data ditambahkan di No. 1")
        print(f"Count: {self.count}")
    
    def insert_posisi(self):
        """2. Insert at given position - timpa index yang dipilih"""
        if self.count >= 10:
            print("Array penuh!")
            return
        
        try:
            no = int(input("No. (1-10): "))
        except ValueError:
            print("No. harus 1-10!")
            return
        
        if no < 1 or no > 10:
            print("No. harus 1-10!")
            return
        
        n = input("NIM: ")
        nm = input("Nama: ")
        
        idx = no - 1
        
        if self.nim[idx] is None:
            self.count += 1
        
        self.nim[idx] = n
        self.nama[idx] = nm
        print(f"Data ditambahkan di No. {no}")
        print(f"Count: {self.count}")
    
    def insert_akhir(self):
        """3. Insert at end - selalu timpa index 9"""
        if self.count >= 10:
            print("Array penuh!")
            return
        
        n = input("NIM: ")
        nm = input("Nama: ")
        
        if self.nim[9] is None:
            self.count += 1
        
        self.nim[9] = n
        self.nama[9] = nm
        print("Data ditambahkan di No. 10")
        print(f"Count: {self.count}")
    
    def delete_awal(self):
        """4. Delete from beginning - hapus index 0"""
        if self.nim[0] is None:
            print("No. 1 kosong!")
            return
        
        print(f"Menghapus No. 1: {self.nim[0]} - {self.nama[0]}")
        self.nim[0] = None
        self.nama[0] = None
        self.count -= 1
        print("Data dihapus")
        print(f"Count: {self.count}")
    
    def delete_posisi(self):
        """5. Delete given position - hapus index yang dipilih"""
        if self.count == 0:
            print("Array kosong!")
            return
        
        try:
            no = int(input("No. yang dihapus (1-10): "))
        except ValueError:
            print("No. harus 1-10!")
            return
        
        if no < 1 or no > 10:
            print("No. harus 1-10!")
            return
        
        idx = no - 1
        
        if self.nim[idx] is None:
            print(f"No. {no} kosong!")
            return
        
        print(f"Menghapus No. {no}: {self.nim[idx]} - {self.nama[idx]}")
        self.nim[idx] = None
        self.nama[idx] = None
        self.count -= 1
        print("Data dihapus")
        print(f"Count: {self.count}")
    
    def delete_akhir(self):
        """6. Delete from end - hapus index 9"""
        if self.nim[9] is None:
            print("No. 10 kosong!")
            return
        
        print(f"Menghapus No. 10: {self.nim[9]} - {self.nama[9]}")
        self.nim[9] = None
        self.nama[9] = None
        self.count -= 1
        print("Data dihapus")
        print(f"Count: {self.count}")
    
    def delete_pertama(self):
        """7. Delete first occurence - cari NIM pertama yang cocok"""
        if self.count == 0:
            print("Array kosong!")
            return
        
        cari = input("Masukkan NIM yang akan dihapus: ")
        
        for i in range(10):
            if self.nim[i] is not None and self.nim[i] == cari:
                print(f"Data ditemukan di No. {i+1}: {self.nim[i]} - {self.nama[i]}")
                self.nim[i] = None
                self.nama[i] = None
                self.count -= 1
                print("Data dihapus")
                print(f"Count: {self.count}")
                return
        
        print("Data tidak ditemukan!")
    
    def show_data(self):
        """8. Show data - tampilkan yang tidak null"""
        print("\n=== DATA MAHASISWA ===")
        print(f"Jumlah: {self.count} data")
        
        if self.count == 0:
            print("Tidak ada data")
            return
        
        print("\nNo.\tNIM\tNama")
        print("------------------------")
        
        for i in range(10):
            if self.nim[i] is not None:
                print(f"{i+1}.\t{self.nim[i]}\t{self.nama[i]}")
    
    def run(self):
        """Main program loop"""
        print("=== PROGRAM ARRAY MAHASISWA ===")
        
        while True:
            print("\n=== MENU ===")
            print("1. Insert at beginning")
            print("2. Insert at given position")
            print("3. Insert at end")
            print("4. Delete from beginning")
            print("5. Delete given position")
            print("6. Delete from end")
            print("7. Delete first occurence")
            print("8. Show data")
            print("9. Exit")
            
            try:
                pilih = int(input("Pilih: "))
            except ValueError:
                print("Pilihan salah!")
                continue
            
            if pilih == 1:
                self.insert_awal()
            elif pilih == 2:
                self.insert_posisi()
            elif pilih == 3:
                self.insert_akhir()
            elif pilih == 4:
                self.delete_awal()
            elif pilih == 5:
                self.delete_posisi()
            elif pilih == 6:
                self.delete_akhir()
            elif pilih == 7:
                self.delete_pertama()
            elif pilih == 8:
                self.show_data()
            elif pilih == 9:
                print("Program selesai")
                break
            else:
                print("Pilihan salah!")


if __name__ == "__main__":
    app = FixedstaticArray()
    app.run()